app/controllers/users/invitations_controller.rb

```diff
@@ -61,6 +67,18 @@ def invited_user_has_no_org?
     invited_user.organisation_id.nil?
   end
 
+  def super_admin?
+    current_user.super_admin?
+  end
+
+  def target_organisation
+    @target_organisation ||= if super_admin?
+                               Organisation.find_by(id: params[:id] || invite_params[:organisation_id])
+                             else
+                               current_user.organisation
```

Comment:
This structure handles both super_admin and non super_admin users, however the method `target_organisation` is only used for super admins.

Makes me think that having a `helper_method` is one extraction too far where a simple instance variable `@target_organisation` would do.

Reply:
moving to using an instance variable would require changes at 2 points.

This contains the branching to one place.

If you'd like to still go down the instance variable route, can you recommend a way, as I'm getting odd behaviour when modifying the `new` and `create` controllers